### ISSUE TITLE
struct.error: 'I' format requires 0 <= number <= 4294967295

### ISSUE THREAD

**Issue by akhti** [ptillet/torch-blocksparse#34]
Hi there!

I'm getting the following error trying to run the simple example:

```.pytb
/home/akhti/.conda/envs/main/lib/python3.7/site-packages/torch_blocksparse-1.1.1-py3.7-linux-x86_64.egg/torch_blocksparse/matmul.py:403: UserWarning: This overload of nonzero is deprecated:
        nonzero()
Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  /pytorch_1595629403081/work/torch/csrc/utils/python_arg_parser.cpp:766.)
  nnz = layout.nonzero()
Traceback (most recent call last):
  File "test.py", line 21, in <module>
    d = softmax(c)
  File "/home/akhti/.conda/envs/main/lib/python3.7/site-packages/torch_blocksparse-1.1.1-py3.7-linux-x86_64.egg/torch_blocksparse/softmax.py", line 361, in __call__
    maxlut, self.bench, time_y)
  File "/home/akhti/.conda/envs/main/lib/python3.7/site-packages/torch_blocksparse-1.1.1-py3.7-linux-x86_64.egg/torch_blocksparse/softmax.py", line 300, in forward
    grid=grid, bench=bench)
  File "/home/akhti/.conda/envs/main/lib/python3.7/site-packages/triton-0.2.3-py3.7-linux-x86_64.egg/triton/kernel.py", line 84, in __call__
    params = pack(self.tys, *[x.data_ptr() if isinstance(x, torch.Tensor) else x for x in args])
struct.error: 'I' format requires 0 <= number <= 4294967295
```

Any thoughts on what could this be?
I'm running on Ubuntu 18.04 with cuda 10.1 and pytorch 1.6

**Comment by ptillet** [comment]
Hey! This is kind of strange. Are you running the exact Python code of the example?

**Comment by akhti** [comment]
> Hey! This is kind of strange. Are you running the exact Python code of the example? I could see this happening when the matrices are very large, i.e., have more than 2^32 nonzero elements.

AFAIU `x.data_ptr()` is the address of the tensor, not size. So it could be huge even for small tensors, e.g.:
```
$ python -c "import torch; print(torch.zeros(10).data_ptr())"
94481101495360
```

**Comment by ptillet** [comment]
Oh yes, I don't think the error is happening with data_ptr(). This is properly encoded as a 64-bit memory address.

`[x.data_ptr() if isinstance(x, torch.Tensor) else x for x in args]`

iterates over all arguments, and my guess is that one of them does not fit in an int. But the example shouldn't have this behavior which is why I'm a bit confused

**Comment by lyn0102** [comment]
meet same question, have you fixed it?

**Comment by ptillet** [comment]
Issue has been fixed in ca1883a959737292a19f7e21b84eecb7c764ce42
